core/nakamoto/mempool_test.go

```diff
@@ -2,8 +2,105 @@ package nakamoto
 
 import (
 	"testing"
+
+	"github.com/liamzebedee/tinychain-go/core"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestMempool(t *testing.T) {
 	// TODO implement.
 }
+
+func isCandidate(t *testing.T, mempool *Mempool, tx *RawTransaction) bool {
+	found := false
+	for _, i := range mempool.candidateTxns {
+		// fmt.Printf("i: %v\n", i)
+		if tx == i {
+			return true
+		}
+	}
+	return found
+}
+
+func TestAddTransaction(t *testing.T) {
```

Comment:
Ok so this tests just adding a tx to the pool, making sure it is inserted into candidate txs set